x-pack/plugins/observability_solution/apm/public/components/shared/transaction_action_menu/sections.ts

```diff
@@ -117,13 +119,14 @@ export const getSections = ({
       label: i18n.translate('xpack.apm.transactionActionMenu.showPodMetricsLinkLabel', {
         defaultMessage: 'Pod metrics',
       }),
-      href: getInfraHref({
-        app: 'metrics',
-        basePath,
-        path: `/link-to/pod-detail/${podId}`,
-        query: infraMetricsQuery,
+      href: assetDetailsLocator?.getRedirectUrl({
+        assetId: podId!,
+        assetType: 'pod',
+        assetDetails: {
+          dateRange: infraMetricsQuery,
```

Comment:
The asset details page doesn't support `pod` type yet - only `hosts` and `containers` are [supported](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/common/asset_details/types.ts) - and the legacy page query param attribute is [`_a`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/public/pages/metrics/metric_detail/hooks/use_metrics_time.ts#L46), so the `dateRange` here will not be propagated to the legacy page.

To avoid making the consumer aware of this detail, we could change the assetDetailsLocator to receive only `assetDetails`, and build and use `_a` for non-supported asset types. wdyt?

Reply:
Seems doable. I'll have a bash at that then.